### PR TITLE
Allow synchronous scanners

### PR DESCRIPTION
**Pull request by Rizato**
Provides `platform.system()` and `platform.machine()` values as fields in AbstractScanner. 

We should only need to call these once when the scanner starts, instead of on every scan to build metadata.

### CODE REVIEW

**Comment by Rizato** [comment]
Cancelled CI on 30d5eaf and d6a6be1 because of later commits

**Comment by Rizato** [comment]
Let me know if there is a known pattern or better way to identify which function it should call.

**Comment by zv** [comment]
blammo!

File: src/polyswarmclient/abstractscanner.py [hunk 1]
```diff
@@ -28,14 +31,65 @@ def __repr__(self):
...
+    This class offers two scan options, which can be specified by passing a ScanMode enum value as `mode`.
+    It uses asynchronous scan by default.
+
+    The function `scan_async` is a coroutine function where everything called from this function must be async compatible. non-blocking,
+    That means it uses only non-blocking IO, and runs nothing cpu-bound, like hash functions.
+
+    The function `scan_sync` is a synchronous function where anything goes.
+    It is called in a ThreadPoolExecutor so it is compatible with the asynch code in the worker.
+
+
```
Comment (alanjds) [inline src/polyswarmclient/abstractscanner.py:55]:
```suggestion
```

File: src/polyswarmclient/abstractscanner.py [hunk 2]
```diff
@@ -46,15 +100,14 @@ async def scan(self, guid, artifact_type, content, metadata, chain):
...
         """
-        pass
+        raise NotImplementedError("Must implement scan_async when using ScanMode.Async")
 
     async def setup(self):
         """Override this method to implement custom setup logic.
 
         This is run immediately after the Scanner class is instantiated and before any calls to the scan() method.
 
         Args:
```
Comment (alanjds) [inline src/polyswarmclient/abstractscanner.py:110]: Consider to remove docstrings on `Args:` if no arg will be described.
```suggestion
        This is run immediately after the Scanner class is instantiated and before any calls to the scan() method.
```

**Pull request merged**